app/Domains/Contact/ManageContact/Web/Controllers/ContactController.php

```diff
@@ -152,7 +152,7 @@ public function destroy(Request $request, int $vaultId, int $contactId)
             'contact_id' => $contactId,
         ];
 
-        (new DestroyContact())->execute($data);
+        DestroyContact::dispatch($data);
```

Comment:
What's your point? Shouldn't we use a queue at all?

Reply:
We should definitely use it but only when the UI takes it into account, like wait for the queue to be executed. In this case the Ui doesn't wait for something to be done in the background so it might lead to errors. For some cases it's not an issue though, like sending emails or provisioning the account.